crates/db_schema/src/impls/password_reset_request.rs

```diff
@@ -52,30 +25,21 @@ impl PasswordResetRequest {
       local_user_id: from_local_user_id,
       token: token_.into(),
     };
-
-    Self::create(pool, &form).await
-  }
-  pub async fn read_from_token(pool: &mut DbPool<'_>, token_: &str) -> Result<Option<Self>, Error> {
     let conn = &mut get_conn(pool).await?;
-    password_reset_request
-      .filter(token.eq(token_))
-      .filter(published.gt(now.into_sql::<Timestamptz>() - 1.days()))
-      .first(conn)
+    insert_into(password_reset_request)
+      .values(form)
+      .get_result::<Self>(conn)
       .await
-      .optional()
   }
 
-  pub async fn get_recent_password_resets_count(
-    pool: &mut DbPool<'_>,
-    user_id: LocalUserId,
-  ) -> Result<i64, Error> {
+  pub async fn read_and_delete(pool: &mut DbPool<'_>, token_: &str) -> Result<Option<Self>, Error> {
     let conn = &mut get_conn(pool).await?;
-    password_reset_request
-      .filter(local_user_id.eq(user_id))
+    delete(password_reset_request)
```

Comment:
Surprised that delete works with `.get_result`, but you verified it below in the tests anyway.